wp-cache-sanitizer.php

```diff
@@ -0,0 +1,726 @@
+      'w3c ',
+      'w3c-',
+      'acs-',
+      'alav',
+      'alca',
+      'amoi',
+      'audi',
+      'avan',
+      'benq',
+      'bird',
+      'blac',
+      'blaz',
+      'brew',
+      'cell',
+      'cldc',
+      'cmd-',
+      'dang',
+      'doco',
+      'eric',
+      'hipt',
+      'htc_',
+      'inno',
+      'ipaq',
+      'ipod',
+      'jigs',
+      'kddi',
+      'keji',
+      'leno',
+      'lg-c',
+      'lg-d',
+      'lg-g',
+      'lge-',
+      'lg/u',
+      'maui',
+      'maxo',
+      'midp',
+      'mits',
+      'mmef',
+      'mobi',
+      'mot-',
+      'moto',
+      'mwbp',
+      'nec-',
+      'newt',
+      'noki',
+      'palm',
+      'pana',
+      'pant',
+      'phil',
+      'play',
+      'port',
+      'prox',
+      'qwap',
+      'sage',
+      'sams',
+      'sany',
+      'sch-',
+      'sec-',
+      'send',
+      'seri',
+      'sgh-',
+      'shar',
+      'sie-',
+      'siem',
+      'smal',
+      'smar',
+      'sony',
+      'sph-',
+      'symb',
+      't-mo',
+      'teli',
+      'tim-',
+      'tosh',
+      'tsm-',
+      'upg1',
+      'upsi',
+      'vk-v',
+      'voda',
+      'wap-',
+      'wapa',
+      'wapi',
+      'wapp',
+      'wapr',
+      'webc',
+      'winw',
+        'winw',
+      'xda ',
+      'xda-'
+  );
+
+    /**
+     * Merges the list of all the options that will act as a list of options allowed to imported
+     *
+     * @since  1.4.4
+     */
+
+  function __construct() {
+
+    $this->allowed_options = apply_filters( 'wp_super_cache_extend_allowed_option', self::ALLOWED_OPTIONS );
+    $this->allowed_booleans = apply_filters( 'wp_super_cache_extend_allowed_booleans', self::ALLOWED_SETTING_BOOLEANS );
+    $this->allowed_integers = apply_filters( 'wp_super_cache_extend_allowed_integers', self::ALLOWED_SETTING_INTS );
+    $this->allowed_strings = apply_filters( 'wp_super_cache_extend_allowed_strings', self::ALLOWED_SETTING_STRINGS );
+
+    $this->allowed_values = array_merge(
+        $this->allowed_options,
+        $this->allowed_booleans,
+        $this->allowed_integers,
+        $this->allowed_strings
+    );
+
+  }
+
+  /**
+   * Determines if the current setting is allowed to imported.  If the current setting is allowed
+   * to imported then perform then sanitize the value correctly.  If the $insert variable is set to
+   * true then this function will call the insert function to write the settings to the wp-cache-config.php
+   * file or to the database, depending on the setting.
+   *
+   * @param String $setting The current setting of the value to be sanitized
+   * @param Mixed $value The value of the current setting being imported
+   * @param Boolean $insert Determines whether the value will be written to the wp-cache-config.php file or inserted into the database
+   *
+   * @since  1.4.4
+   */
+  function sanitize( $setting, $value, $insert = false ) {
+
+    if ( ! in_array( $setting, $this->allowed_values ) )
```

Comment:
We already have to check each array individually, so we may as well not create `$this->allowed_values` at all and do `else return false;` below.